src/Shared/CertificateGeneration/CertificateManager.cs

```diff
@@ -345,9 +393,56 @@ internal void ExportCertificate(X509Certificate2 certificate, string path, bool
             }
 
             byte[] bytes;
+            byte[] keyBytes;
+            byte[] pemEnvelope = null;
             try
             {
-                bytes = includePrivateKey ? certificate.Export(X509ContentType.Pkcs12, password) : certificate.Export(X509ContentType.Cert);
+                if (includePrivateKey)
+                {
+                    switch (format)
+                    {
+                        case CertificateKeyExportFormat.Pfx:
+                            bytes = includePrivateKey ? certificate.Export(X509ContentType.Pkcs12, password) : certificate.Export(X509ContentType.Cert);
```

Comment:
`includePrivateKey` is always `true` at this point. Maybe check if `password != null`.